homeassistant/components/google_assistant/trait.py

```diff
@@ -901,31 +907,85 @@ class TemperatureControlTrait(_Trait):
 
     name = TRAIT_TEMPERATURE_CONTROL
 
+    commands = [
+        COMMAND_SET_TEMPERATURE,
+    ]
+
     @staticmethod
     def supported(domain, features, device_class, _):
         """Test if state is supported."""
         return (
             domain == sensor.DOMAIN
             and device_class == sensor.SensorDeviceClass.TEMPERATURE
+        ) or (
+            domain == water_heater.DOMAIN
+            and features & WaterHeaterEntityFeature.TARGET_TEMPERATURE
```

Comment:
Until now, `TemperatureControlTrait` has only been used for temperature sensors, and we also mention this in the docstring:
```py
    """Trait for devices (other than thermostats) that support controlling temperature. Workaround for Temperature sensors.
    https://developers.google.com/assistant/smarthome/traits/temperaturecontrol
    """
```

We should rewrite the doscstring since this is no longer true, maybe:
```py
    """Trait for devices (other than thermostats) that support controlling temperature.

    Also used for temperature sensors.
    https://developers.google.com/assistant/smarthome/traits/temperaturecontrol
    """
```

Reply:
It is still a work-a-round for temperature sensors.

Reply:
Updated the docstr though